UI redesign : change font & color of timestamp in a message

supriya-045 opened the pull request:
Fixes #21752 

Made the required changes in  static/styles/zulip.css
and static/styles/dark_theme.css in issue #21752

<!-- What's this PR for?  (Just a link to an issue is fine.) -->https://github.com/zulip/zulip/issues/21752

**Testing plan:** <!-- How have you tested? -->By running  locally after changes 

**GIFs or screenshots:** 
  https://zulip.re
![light_before](https://user-images.githubusercontent.com/75739890/162609585-d2e6fa60-fd91-4fed-9311-a380299abeba.png)

![final light](https://user-images.githubusercontent.com/75739890/163688138-d1bb775d-1053-486f-8c3c-0465db594f96.png)




In dark mode 
![beforedarknew](https://user-images.githubusercontent.com/75739890/163530899-5c1b192a-e7df-48a4-be86-d84c3ebe48ab.png)


![finaldark](https://user-images.githubusercontent.com/75739890/163688173-f5a41bc9-5b6d-4207-9671-0ee45a9802e4.png)













<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

alya commented:
Thanks @supriya-045 ! In addition to addressing the feedback above and posting an updated screenshot, please also edit your commit message according to https://zulip.readthedocs.io/en/latest/contributing/version-control.html.

You can post and to request a review when ready. :)

supriya-045 commented:
> Thanks @supriya-045 ! In addition to addressing the feedback above and posting an updated screenshot, please also edit your commit message according to https://zulip.readthedocs.io/en/latest/contributing/version-control.html.
> 
> You can post and to request a review when ready and all the tests are passing. :)

All checks are passed now,please review the PR.

alya commented:
@terpimost would you like to take a look?

In the screenshot, the font looks a bit funny to me next to the date font in the message header, but I don't know if we think that's OK for now?

terpimost commented:
If that would be easy to update the date in the header of the topic lets do that.
Light theme:  color is `hsla(0, 0%, 37%, 1)` and
```css
font-size: 15px;
line-height: 18px;
text-align: right;
font-variant: small-caps;
font-weight: 400;
```
and lets also do `font-variant: all-small-caps;` as @sayamsamal suggested.

For the dark theme the color is `background: hsla(0, 0%, 70%, 1);`

supriya-045 commented:
> If that would be easy to update the date in the header of the topic lets do that. Light theme: color is `hsla(0, 0%, 37%, 1)` and
> 
> ```css
> font-size: 15px;
> line-height: 18px;
> text-align: right;
> font-variant: small-caps;
> font-weight: 400;
> ```
> 
> and lets also do `font-variant: all-small-caps;` as @sayamsamal suggested.
> 
> For the dark theme the color is `background: hsla(0, 0%, 70%, 1);`

Sir 

> If that would be easy to update the date in the header of the topic lets do that. Light theme: color is `hsla(0, 0%, 37%, 1)` and
> 
> ```css
> font-size: 15px;
> line-height: 18px;
> text-align: right;
> font-variant: small-caps;
> font-weight: 400;
> ```
> 
> and lets also do `font-variant: all-small-caps;` as @sayamsamal suggested.
> 
> For the dark theme the color is `background: hsla(0, 0%, 70%, 1);`

As per the issue of changing the font and color of timestamp also do i need to update the date font in the header?

alya commented:
@supriya-045  Right, @terpimost 's and my feedback above is that we would like to make the header changes at the same time. Please add those to your PR (in a separate commit), or let us know if you aren't able to do so. Thanks!

alya commented:
@supriya-045 please post a comment if/when this is ready for another review. Thanks!

supriya-045 commented:
> @supriya-045 please post a comment if/when this is ready for another review. Thanks!

@alya and @terpimost I have made the  changes in the style of date in message header ,please review it. Thanks!

terpimost commented:
@supriya-045 could you please attach light and dark theme screen shots?

supriya-045 commented:
> Made the required changes in static/styles/zulip.css and static/styles/dark_theme.css in issue #21752
> https://github.com/[/issues/21752](https://github.com/zulip/zulip/issues/21752)
> 
> **Testing plan:** By running locally after changes
> 
> **GIFs or screenshots:** https://zulip.re ![light_before](https://user-images.githubusercontent.com/75739890/162609585-d2e6fa60-fd91-4fed-9311-a380299abeba.png)
> 
> ![all small caps](https://user-images.githubusercontent.com/75739890/162996766-3a313859-fa3f-430f-bba2-c54c82b09e57.png)
> 
> ![message header](https://user-images.githubusercontent.com/75739890/163256820-71a092e6-86fc-4e48-922b-787adecc728e.png)



> @supriya-045 could you please attach light and dark theme screen shots?

@terpimost I have added the screenshot of dark mode too,please review it.Thanks!

supriya-045 commented:
> @supriya-045 could you please attach light and dark theme screen shots?

@terpimost I have added the screenshot of dark mode too, please review it. Thanks!

supriya-045 commented:
> > @supriya-045 could you please attach light and dark theme screen shots?
> 
> @alya @terpimost I have added the screenshot of dark mode too, please review it. Thanks!

terpimost commented:
@supriya-045 thank you for the screenshots. It looks like that in the current design the date is too dark in the dark mode.

supriya-045 commented:
> @supriya-045 thank you for the screenshots. It looks like that in the current design the date is too dark in the dark mode.

I have applied the same as suggested by you , should I use same as before?

terpimost commented:
Let me check on my side and get back to you

On Fri, Apr 15, 2022 at 3:43 PM Supriya Patidar ***@***.***>
wrote:

> @supriya-045 <https://github.com/supriya-045> thank you for the
> screenshots. It looks like that in the current design the date is too dark
> in the dark mode.
>
> I have applied the same as suggested by you , should I use same as before?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/zulip/zulip/pull/21754#issuecomment-1100321668>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAOQVTNTXT5RAJBAEQ4R7XTVFHBE5ANCNFSM5TAPUY2A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

supriya-045 commented:
> Let me check on my side and get back to you
> […](#)
> On Fri, Apr 15, 2022 at 3:43 PM Supriya Patidar ***@***.***> wrote: @supriya-045 <https://github.com/supriya-045> thank you for the screenshots. It looks like that in the current design the date is too dark in the dark mode. I have applied the same as suggested by you , should I use same as before? — Reply to this email directly, view it on GitHub <[#21754 (comment)](https://github.com/zulip/zulip/pull/21754#issuecomment-1100321668)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAOQVTNTXT5RAJBAEQ4R7XTVFHBE5ANCNFSM5TAPUY2A> . You are receiving this because you were mentioned.Message ID: ***@***.***>

@terpimost @alya As I have made the PR for this issue ,can I unassign myself because due to this I am unable to claim another issues. And please let me know if it require some changes . Thanks!

terpimost commented:
@supriya-045 sorry, for confusing instructions. 
Lets try that again. For the dark theme the timestamp color is `hsla(0, 0%, 50%, 1);`
and for the date color is `hsla(0, 0%, 70%, 1);`
Please check opacity (transparency) of parenting containers. Since the date is lighter of a color it should be visible better.
Here is the part from the design:
<img width="349" alt="image" src="https://user-images.githubusercontent.com/1903309/163680982-77d97e30-6162-46fe-b404-3cb7b59b6883.png">

supriya-045 commented:
> @supriya-045 sorry, for confusing instructions. Lets try that again. For the dark theme the timestamp color is `hsla(0, 0%, 50%, 1);` and for the date color is `hsla(0, 0%, 70%, 1);` Please check opacity (transparency) of parenting containers. Since the date is lighter of a color it should be visible better. Here is the part from the design: <img alt="image" width="349" src="https://user-images.githubusercontent.com/1903309/163680982-77d97e30-6162-46fe-b404-3cb7b59b6883.png">

@terpimost I have done the changes suggested by you and added the screenshot , please review it.Thanks!

terpimost commented:
Great. I would love to test deploy it and see if anything should be tined further. @alya how should we move further with these improvements by @supriya-045 ?

supriya-045 commented:
> Great. I would love to test deploy it and see if anything should be tined further. @alya how should we move further with these improvements by @supriya-045 ?

@terpimost @alya As I have made the PR for this issue ,can I unassign myself because of  this I am unable to claim another issues. And please let me know if it require some changes . Thanks!

alya commented:
Hm, testing this on dev, the alignment looks off to me -- the date is a bit lower than the other text in the topic bar.

<img width="660" alt="new_router_is_skipping_quickly_-_Zulip_Dev_-_Zulip" src="https://user-images.githubusercontent.com/2090066/164166991-0aa956d0-7a45-43c5-b373-9f2745e189a9.png">

alya commented:
The alignment of the timestamp seems too high compared to the other icons; I'm not sure about the sender.
<img width="233" alt="Screen Shot 2022-04-19 at 11 47 58 PM" src="https://user-images.githubusercontent.com/2090066/164167619-1a03ad26-5694-413f-8e8f-887210deb1e8.png">

<img width="1000" alt="Screen Shot 2022-04-19 at 11 48 30 PM" src="https://user-images.githubusercontent.com/2090066/164167671-4d221836-04b9-4488-a41f-4f8d420f86de.png">

supriya-045 commented:
> Hm, testing this on dev, the alignment looks off to me. The date is a bit lower than the other text in the topic bar, and I think it would look better if that were not the case.
> 
> <img alt="new_router_is_skipping_quickly_-_Zulip_Dev_-_Zulip" width="660" src="https://user-images.githubusercontent.com/2090066/164166991-0aa956d0-7a45-43c5-b373-9f2745e189a9.png">

Alignment is now same.
![dark date alignment](https://user-images.githubusercontent.com/75739890/164184241-d84f3662-c7bf-4b7e-a3e5-69410caf8569.png)



![light alignment](https://user-images.githubusercontent.com/75739890/164184266-bb829d02-e4e6-4b4c-9257-1e5fffd4f2c0.png)

supriya-045 commented:
> The alignment of the timestamp seems too high compared to the other icons; I'm not sure about the sender. <img alt="Screen Shot 2022-04-19 at 11 47 58 PM" width="233" src="https://user-images.githubusercontent.com/2090066/164167619-1a03ad26-5694-413f-8e8f-887210deb1e8.png">
> 
> <img alt="Screen Shot 2022-04-19 at 11 48 30 PM" width="1000" src="https://user-images.githubusercontent.com/2090066/164167671-4d221836-04b9-4488-a41f-4f8d420f86de.png">


Alignment of timestamp is now same as other icons ,please review it. Thanks
![alignmentindark](https://user-images.githubusercontent.com/75739890/164185405-d75c3193-1dee-4ba4-ad1f-ed5bc614db5a.png)
![alignmentwhite](https://user-images.githubusercontent.com/75739890/164185436-80639d42-3d47-45f5-9579-f91e4f0cb62a.png)

terpimost commented:
@supriya-045 and @alya thank you so much for noticing and working on such details.

alya commented:
@supriya-045 please update your commit messages according to https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages

Thanks!

alya commented:
@terpimost is this otherwise ready for testing on CZO from your perspective?

medhatiwari commented:
> @supriya-045 please update your commit messages according to https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages
> 
> Thanks!
Hey, I have updated the commit message.

supriya-045 commented:
> @supriya-045 please update your commit messages according to https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages
> 
> Thanks!

Hey! I have changed the commit message.

terpimost commented:
> @terpimost is this otherwise ready for testing on CZO from your perspective?

Yes

alya commented:
@timabbott could you please evaluate this for test-deployment on CZO? @terpimost and I have reviewed it.

timabbott commented:
I've squashed the commits and cleaned up some stale comments in this PR. I think we can potentially test-deploy it, but I noted a few issues:
* Hovering over the timestamp results in a slow 1.5s transition/fade towards blue link styling. This is likely just stale `transition` CSS from some previous design that needs to be excised (along with the likely misplaced comments explaining them).
* I'm suspicious that the vertical alignment was not fixed correctly; it seems possible that, for example, the timestamp was actually aligned with the sender name (on the left) but that . So we should check that.
* I note that the issue description has `hsla(0, 0%, 60%, 1);` vs. the 50% that this implements; based on https://github.com/zulip/zulip/pull/21754#issuecomment-1100689525 I think the issue description is wrong, but just flagging this.
* Ideally, we'd use 110% for the font size variations (rather than 15px), since that's the way to make it work when we support different font sizes. We should try to use that pattern wherever we're doing something that's intended to be a different size from the rest of the app's content.
* I'm not sure about the maintainability of using a custom text color for this. Previously, this component had the same font color as the rest of the app, with the idea that we'd just have a common default text font color, and components that just want a lighter/darker variant can use opacity for it. (The 50% used before this was crazy, though -- one usually wants 70% or 80%). So @terpimost I wanted to ask if the colors you selected here are equivalent to something like 70% opacity on what would be the future default font color for the app; if so, we can define `$redesign_font_color_light` and `$redesign_font_color_dark` variables and use those. The other advantage of opacity for the job is that often a single value will look OK in both light and dark themes. 

To be clear, I'm fine with testing and merging this one with the current approach, but I figure this PR is a good opportunity to decide those two "how we write CSS for the redesign" questions, since they're most of the code of this PR, and it'll smooth future ones if we make a policy decision on how we want to write this stuff.

In any case, @amanagr perhaps you can make a new PR for this issue, debug the blue link styling thing (perhaps have just a new prep commit removing it -- I checked it's there in `main` but just really subtle), and perhaps action whatever we conclude with @terpimost on the percentage/opacity questions. It might be easier for you to understand the plan to just read the issue thread and this comment and then try to do the issue yourself, and then compare with this PR's code, since the comment thread here is so long.

terpimost commented:
@timabbott yes, we can make text for date as regular text with 75% opacity:
<img width="1367" alt="image" src="https://user-images.githubusercontent.com/1903309/182747183-6ba90338-216c-4abc-983a-a63083499fa3.png">
<img width="1416" alt="image" src="https://user-images.githubusercontent.com/1903309/182747243-58b17337-a409-4483-9e69-6da7b0e94f85.png">

timabbott commented:
Closing in favor of https://github.com/zulip/zulip/pull/22661.
